log-collector/src/main/scala/io/phdata/pulse/logcollector/LogCollector.scala

```diff
@@ -53,24 +56,48 @@ object LogCollector extends LazyLogging {
 
     val routes = new LogCollectorRoutes(solrService)
 
-    // Starts Http Service
-    def start(port: Int): Future[Unit] =
-      Http().bindAndHandle(routes.routes, "0.0.0.0", port = port)(materializer) map { binding =>
+    def serve(port: Int): Unit = {
+      // Starts Http Service
+      val httpServerFuture = Http().bindAndHandle(routes.routes, "0.0.0.0", port = port)(
+        materializer) map { binding =>
         logger.info(s"Log Collector interface bound to: ${binding.localAddress}")
       }
 
-    val s = start(cliParser.port())
+      httpServerFuture.onComplete {
+        case Success(v) => ()
+        case Failure(e) => throw new RuntimeException(e)
+      }
 
-    s.onComplete {
-      case Success(v) => ()
-      case Failure(e) => throw new RuntimeException(e)
+      // Start LogCollector HttpService
+      Await.ready(
+        httpServerFuture,
+        Duration.Inf
+      )
     }
 
-    // Start LogCollector HttpService
-    Await.ready(
-      s,
-      Duration.Inf
-    )
+    cliParser.mode() match {
+      case "kafka" => {
+        consume(solrCloudStream, cliParser.zkHosts(), cliParser.port(), cliParser.topic())
+      }
+      case _ => {
+        serve(cliParser.port())
+      }
+    }
   }
 
+  // Starts Kafka Consumer
+  def consume(solrCloudStream: SolrCloudStream, zkHost: String, port: Int, topic: String): Unit = {
+    val kafkaConsumer      = new PulseKafkaConsumer(solrCloudStream)
+    val kafkaConsumerProps = new Properties()
+
+    kafkaConsumerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, s"$zkHost:$port")
+    kafkaConsumerProps.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+                           "org.apache.kafka.common.serialization.StringDeserializer")
+    kafkaConsumerProps.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+                           "org.apache.kafka.common.serialization.StringDeserializer")
+    kafkaConsumerProps.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
+    kafkaConsumerProps.put(ConsumerConfig.GROUP_ID_CONFIG, "pulse-kafka")
```

Comment:
How about using something like `pulse-log-collector` for the group name. It'll be more specific.